### PR TITLE
Support for terragrunt files

### PR DESCRIPTION
**Pull request by gmcoringa**
Add support for reading terragrunt.hcl file.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -415,3 +431,23 @@ func installTFProvidedModule(dir string, custBinPath *string) {
...
 }
+
+// Install using version constraint from terragrunt file
+func installTGHclFile(tgFile *string, custBinPath *string) {
+	fmt.Printf("Terragrunt file found: %s\n", *tgFile)
+	content := retrieveFileContents(*tgFile)
+	var constraint = ""
+
+	for _, line := range strings.Split(content, "\n") {
+		regex, _ := regexp.Compile(`^terraform_version_constraint\s+=\s+"(?P<version>.*)".*`)
```
Comment (versus) [inline main.go:442]: I think you should move regexp.Compile()  before loop

Comment (gmcoringa) [reply main.go]: fixed

**Comment by warrensbox** [comment]
This is merged to master but not released yet. I am still testing.
One thing, I would like is to use the HCL parser instead of regex. 
See: https://github.com/yutannihilation/go-parse-hcl-example/blob/master/main.go
The hcl parser would take parse `terraform_version_constraint` from the terragrunt file.

**Comment by warrensbox** [comment]
@gmcoringa  your changes are released with version 0.12
Instead of using regex, I refactored your code to use the hcl parser:
https://github.com/warrensbox/terraform-switcher/blob/2865ec9eb81fa8dc8cac1804d9c946999e1a785e/main.go#L441
Release #163 
Please upgrade and verify:
https://tfswitch.warrensbox.com/Upgrade-or-Uninstall/

**Pull request merged**